### ISSUE TITLE
Error when naming a bookmark "shift"

### ISSUE THREAD

**Issue by alinnert** [DeMille/bookmark#2]
I've tried to create a bookmark named "shift". Since then some commands stopped working.

The bookmark was created successfully. I can see that by typing `bm ls`. (this still works)

But `bm --help` creates an error:

~~~
find: '/n/Abteilungsleitung': Permission denied
~~~

When I try to remove that bookmark by typing `bm rm shift` I get a bit more detailed error:

~~~
find: '/n/Abteilungsleitung': Permission denied
module.js:472
    throw err;
    ^

Error: Cannot find module 'C:\Users\linnert\AppData\Roaming\npm\node_modules\bookmark\cli\cli-shift'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:422:7)
    at startup (bootstrap_node.js:143:9)
    at bootstrap_node.js:537:3
~~~

I have no idea where this `Abteilungsleitung` is coming from. It's a German word, but I can't find a folder or file that's named like this nor can I fand this word in your source code...

**UPDATE:**

Okay, I found a folder with the name "Abteilungsleitung". But it's behind a symlink that's pointing to a network drive. Why does bookmark stumble over that directory? No action I did was related to that symlink.

**Comment by DeMille** [comment]
Ha! That's a pretty funny error.

The first part was an easy fix. It stems from an old bug with commander.js where it did something like:
```js
if (array[user_input]) doThing();
```
So any `user_input` that is array method (like shift, pop, push, etc) trips up the check, causing commander to try to do a git style sub command. This was fixed in newer versions of commander, so I just updated the dependencies. Updating bookmark should fix that issue.

I'm not completely sure about the `Abteilungsleitung` part, but its probably from the `find` call in the [batch file](https://github.com/DeMille/bookmark/blob/master/shell/bm.bat#L120) where its _supposed_ to be running the node script and returning the output. Does it throw that error on any input other than 'shift'?

**Comment by alinnert** [comment]
Oh, I see. That's interesting.

Yes, the error also occurs with `bm --help`. But it doesn't occur when I use `bm <bookmark_name>`, `bm add` or `bm ls`.

**Comment by DeMille** [comment]
Huh, yeah those all have the same `find` syntax.

The only weird guess I could throw out is if the wrong find command is getting used. The dos find command is only for finding substrings within text but the unix find command looks for files. Do you have cygwin and the unix find command on your path or something? Maybe that would explain why it's looking at the filesystem.

Other than that I'm not really sure where that's coming from, sorry :(

**Comment by alinnert** [comment]
Oh, yes. That's it. I'm using Cmder (basically ConEmu) + Git for Windows which includes cygwin. It works fine on the default Windows prompt.

But now I'm a little confused. Where is `bookmark` supposed to work? Windows only? But version 0.2.1 already fixed an issue related to cygwin.

**Comment by DeMille** [comment]
Oh weird, that's my setup too, but my path ordering must be different somewhere. My dos `find` has priority so I've never realized that might be a problem. Ideally this should work everywhere, plain cmd, cygwin, osx, etc.

It's an easy fix though, I just changed the `find` sections to use a `findstr` instead. `findstr` is a dos only command so there shouldn't be an issues with overlap.

I just pushed it the changed to npm, let me know if it works for you?
